staging/src/k8s.io/client-go/discovery/discovery_client.go

```diff
@@ -145,9 +145,9 @@ func (d *DiscoveryClient) ServerGroups() (apiGroupList *metav1.APIGroupList, err
 		apiGroupList = &metav1.APIGroupList{}
 	}
 
-	// append the group retrieved from /api to the list if not empty
+	// prepend the group retrieved from /api to the list if not empty
 	if len(v.Versions) != 0 {
-		apiGroupList.Groups = append(apiGroupList.Groups, apiGroup)
+		apiGroupList.Groups = append([]metav1.APIGroup{apiGroup}, apiGroupList.Groups...)
```

Comment:
am a bit surprised that this is client side. Have you check what comes from the actual discovery endpoint? Is the order correct there?

Reply:
there are two discovery endpoints, `/api` and `/apis`. this stitches them together